File: Master_the_Key_ROS_Concepts/Lec1_Create_a_Cakin_Workspace/catkin_ws/src/my_number_counter/src/number_publisher.cpp
#include <ros/ros.h>
#include <std_msgs/Int64.h>

int main(int argc, char **argv)
{
	ros::init(argc, argv, "number_publisher", ros::init_options::AnonymousName);

	ros::NodeHandle nh;

	ROS_INFO("Node has been started: ");

	ros::Publisher pub = nh.advertise<std_msgs::Int64>("/number",10);
	
	//[rosparam] 
	double publish_frequency;
	nh.getParam("/number_publish_frequency", publish_frequency);
	ros::Rate rate(publish_frequency);
	//ros::Rate rate(3);
	
	//[rosparam]
	int number;
	nh.getParam("/number_to_publish", number);
	
	nh.setParam("/just_another_param", "Bye");

	std_msgs::Int64 msg;
	msg.data = -1;
	while(ros::ok()){
		pub.publish(msg);
		msg.data= number;
		//msg.data+=1;
		rate.sleep();
	}

}

File: Master_the_Key_ROS_Concepts/Lec1_Create_a_Cakin_Workspace/catkin_ws/src/my_number_counter/scripts/number_publisher.py
#!/usr/bin/env python3

import rospy
from std_msgs.msg import Int64

if __name__ == '__main__':
    rospy.init_node('number_publisher', anonymous=True)

    pub = rospy.Publisher("/number", Int64, queue_size=10)
    
    # [rosparam] ex:publish_frequency = rospy.get_param("/nameofparam")
    publish_frequency = rospy.get_param("/number_publish_frequency")
    rate = rospy.Rate(publish_frequency)

    
    #rate = rospy.Rate(2)
    
    # [rosparam] for msg.data
    number = rospy.get_param("/number_to_publish")

    # [rosparam] We can also set param here
    rospy.set_param("/another_param", "Hello")

    msg = Int64()
    msg.data = -1
    while not rospy.is_shutdown():
        pub.publish(msg)
        #msg.data += 1
        msg.data = number
        rate.sleep()
        
    rospy.loginfo("Node was stopped")
